maas/resource_maas_subnet_ip_range_test.go

```diff
@@ -0,0 +1,186 @@
+		resource.TestCheckResourceAttr(ipRangeAttrName, "comment", comment),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "start_ip", ipStart),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "end_ip", ipEnd),
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccCheckMAASSubnetIPRangeDestroy,
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			// Test creation
+			{
+				Config: testAccSubnetIPRangeExampleResource(
+					subnet_name,
+					range_type,
+					comment,
+					ipStart,
+					ipEnd,
+				),
+				Check: resource.ComposeTestCheckFunc(checks...),
+			},
+			// Test if resource drift is detected by modifying the IP range using the
+			// GO MAAS client, and then checking if the state is updated correctly on
+			// refresh
+			{
+				PreConfig: func() {
+					client := testutils.TestAccProvider.Meta().(*maas.ClientConfig).Client
+					params := entity.IPRangeParams{
+						Type:    range_type,
+						Comment: commentMod,
+						StartIP: ipStartMod,
+						EndIP:   ipEndMod,
+						Subnet:  strconv.Itoa(ipRange.Subnet.ID),
+					}
+					// Update the IP range to changed values that should be read into state
+					_, err := client.IPRange.Update(ipRange.ID, &params)
+					if err != nil {
+						t.Fatalf("Failed to update IP range: %s", err)
+					}
+				},
+				RefreshState:       true,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(ipRangeAttrName, "start_ip", ipStartMod),
+				),
+			},
+			// Test import
+			{
+				ResourceName:      ipRangeAttrName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+// Check if the IP range specified actually exists in MAAS
+func testAccMAASSubnetIPRangeCheckExists(rn string, ipRange *entity.IPRange) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("resource not found: %s\n %#v", rn, s.RootModule().Resources)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("resource id not set")
+		}
+
+		conn := testutils.TestAccProvider.Meta().(*maas.ClientConfig).Client
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		gotIPRange, err := conn.IPRange.Get(id)
+		if err != nil {
+			return fmt.Errorf("error getting ip range: %s", err)
+		}
+
+		*ipRange = *gotIPRange
+
+		return nil
+	}
+}
+
+// An example resource configuration for a subnet IP range.
+// Note that a subnet is required to create an IP range.
+func testAccSubnetIPRangeExampleResource(
+	name_subnet string,
+	range_type string,
+	comment string,
+	start_ip string,
+	end_ip string,
+) string {
+	// A subnet is required to create an IP range
+	return fmt.Sprintf(`
+		resource "maas_subnet" "test_subnet" { 
```

Comment:
This is still not properly formatted Terraform code. For example, gateway_ip and dns_servers `=` are not aligned with `=` symbols of the other fields. In addition, the valid Terraform indentation is 2 spaces. Finally, the string should start at the beginning of the line and can disrespect the Go indentation due to the usage of `fmt.Sprintf`. My advice for your future tests is to first write your Terraform in a .tf file, then ask your Terraform binary or IDE to format with `terraform fmt`, then paste inside the acceptance test function.